lib/Dialect/XTenNN/IR/XTenNNOps.cpp

```diff
@@ -297,6 +320,64 @@ LogicalResult amd::xten_nn::GroupConv2dOp::verify() {
   return success();
 }
 
+LogicalResult amd::xten_nn::GroupQuantizeOp::verify() {
+  auto inputShape = cast<ShapedType>(getInput().getType()).getShape();
+  auto scalesShape = cast<ShapedType>(getScales().getType()).getShape();
+  auto zerosShape = cast<ShapedType>(getZeros().getType()).getShape();
+  auto quantsShape = cast<ShapedType>(getQuants().getType()).getShape();
+
+  if (inputShape != quantsShape) {
+    return emitOpError() << "input and quants must have the same shape (" << inputShape << " v " << quantsShape << ")";
+  }
+
+  if (scalesShape != zerosShape) {
+    return emitOpError() << "scales and zeros must have the same shape (" << scalesShape << " v " << zerosShape << ")";
+  }
+
+  if (scalesShape.back() != 1) {
+    return emitOpError() << "groups needs to be expressed in the innermost dimension of scales vs quants (" << scalesShape.back() << ")" ;
```

Comment:
I think it would be useful to describe this in the operation definition.

Reply:
I was just thinking along the lines of a comment in the operation documentation. But we can leave it for now.